### DISCUSSION TITLE
Thoughts about jump and surround

### DISCUSSION THREAD

**Discussion by fecet** [ayamir/nvimdots#773] [General]
https://github.com/phaazon/hop.nvim has been inactive for a while and the author just make an announcement in [twin](https://this-week-in-neovim.org/2023/May/29). 

Of course, this cannot be a reason to retire hop.nvim, as it is currently functioning well for me. However, I have always been interested in alternatives such as leap.nvim, and I would like to hear everyone's thoughts on this.

Another topic I would like to discuss is the surround/textobject plugins, as they often have a logical connection in terms of operation like https://github.com/mfussenegger/nvim-treehopper, and both of these plugins prefer key mappings starting with `s` so it would be best to configure them together. It seems that our repository is currently lacking such plugins, so could anyone share their best practices with them?

For me personally, I am accustomed to performing surround operations after entering visual selection. However, such plugins are rare, and the [one](https://github.com/ur4ltz/surround.nvim) I am currently using only provides adding functionality without deletion/replacement. On the other hand, I use the incremental selection feature of LSP to determine the area, but it doesn't always handle parentheses [ideally](https://www.reddit.com/r/neovim/comments/13nohrl/how_to_use_incremental_selection_to_select_a_word), which often trips me up.

**Comment by ayamir** [comment]
The way I uses `nvim-treehopper` is motion + `m` + selection like what I showed in wiki and [youtube](https://www.youtube.com/watch?v=4esdUMHXNTo).

**Reply by fecet** [reply]
Thanks for sharing! Treehopper is also a elegant way to select but is there anyway to do surround operations like add, remove, replace after that? It would be a litter inconsistet if we have a fast wat to select a textobject, say a large dict, but have to do `saw)`(or something like) in normal mode.
